File: src/main/kotlin/fizzbuzz/FizzBuzzConverter.kt
package fizzbuzz

import fizzbuzz.checker.BuzzChecker
import fizzbuzz.checker.FizzChecker

class FizzBuzzConverter(
    private val fizzChecker: FizzChecker,
    private val buzzChecker: BuzzChecker
) {
    fun convertNumber(number: Int): String {
        return when {
            fizzChecker.isFizz(number) && buzzChecker.isBuzz(number) -> "FizzBuzz"
            fizzChecker.isFizz(number) -> "Fizz"
            buzzChecker.isBuzz(number) -> "Buzz"
            else -> number.toString()
        }
    }
}

File: SuperiorFizzBuzz/SuperiorFeesBush/FizzBuzzConverter.cs
public static class FizzBuzzConverter
{
    public static string Convert(int number)
    {
        if (IsFizz(number) && IsBuzz(number)) return "FizzBuzz";
        else if (IsFizz(number)) return "Fizz";
        else if (IsBuzz(number)) return "Buzz";
        else return null;
    }

    private static bool IsFizz(int number)
    {
        return number % 5 == 0;
    }

    private static bool IsBuzz(int number)
    {
        return number % 3 == 0;
    }
}